### ISSUE TITLE
Merging leg after leg after leg...

### ISSUE THREAD

**Issue by JustinAiken** [adhearsion/adhearsion#378]
- A leg calls B leg
- B leg places A leg on hold
- B leg calls C leg
- B leg wants to connect all parties - merges A->B Dial to B->C Dial
- B leg hangs up
- C leg dials D leg
- C leg wants to connect all parties ->  
  Merges the C->D Dial to either the A->B dial or B->C dial (or vica-versa)

Whenever this merge happens, I end up with my original A->B Dial getting unblocked and hungup, while D leg sits idly there alone..

It also crashes [here](https://github.com/adhearsion/adhearsion/blob/develop/lib/adhearsion/call_controller/dial.rb#L196) unless you add `:internal_server_error`  to the [guard method](https://github.com/adhearsion/adhearsion/blob/develop/lib/adhearsion/call_controller/dial.rb#L341)

I've gotten a log both ways here:

[Reproduction app and logs](https://gist.github.com/JustinAiken/8e41552b88103819834c)
